### DISCUSSION TITLE
CosmosOS and 16GB RAM?

### DISCUSSION THREAD

**Discussion by Shlepsss** [CosmosOS/Cosmos#2849] [General]
When im trying to start OS with more than 4GB RAM, VBECanvas and Canvas don't working on real hardware and VMWare. How to make my OS work with 8, 16, 32 GB RAM?

**Comment by retrievaaaa** [comment]
Cosmos is 32 bit, running a 32-bit OS with more than 4 GB is physically impossible.

**Reply by jwoff78** [reply]
it is with Physical Address Extension 'PAE' but I have no clue how to write a driver for that

**Reply by retrievaaaa** [reply]
Fair but there is no use to it without paging (well, you need paging to use it to begin with) or user space. The addressable memory at a time stays at 4GB.

**Reply by Shlepsss** [reply]
Cosmos now work with Limine bootloader. Is possible to make x64?

**Reply by jwoff78** [reply]
That's already in progress but considering how hard coded IL2CPU is for x86 it will take a long time
